File: handler.js
require('dotenv').config();


const nodemailer = require("nodemailer");

module.exports.sendEmail = async (event) => {
  try {
    const { receiver_email, subject, body_text } = JSON.parse(event.body || "{}");

    // Input validation
    if (!receiver_email || !subject || !body_text) {
      return {
        statusCode: 400,
        body: JSON.stringify({ message: "Missing required fields" }),
      };
    }

    // Configure transporter
    const transporter = nodemailer.createTransport({
      service: "gmail",
      auth: {
        user: process.env.SENDER_EMAIL,
        pass: process.env.SENDER_PASSWORD,
      },
    });

    // Send email
    await transporter.sendMail({
      from: process.env.SENDER_EMAIL,
      to: receiver_email,
      subject: subject,
      text: body_text,
    });

    return {
      statusCode: 200,
      body: JSON.stringify({ message: "Email sent successfully!" }),
    };
  } catch (error) {
    console.error("Error sending email:", error);
    return {
      statusCode: 500,
      body: JSON.stringify({
        message: "Failed to send email",
        error: error.message,
      }),
    };
  }
};

File: handler.py
import json

def send_email(event, context):
    try:
        body = json.loads(event.get('body', '{}'))
        receiver_email = body.get('receiver_email')
        subject = body.get('subject')
        body_text = body.get('body_text')

        if not all([receiver_email, subject, body_text]):
            return {
                "statusCode": 400,
                "body": json.dumps({"error": "Missing required fields"})
            }

        # Mock sending email (for offline testing)
        print(f"Email would be sent to {receiver_email} with subject '{subject}' and body '{body_text}'")

        return {
            "statusCode": 200,
            "body": json.dumps({"message": "Email sent successfully (mock)"})
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)})
        }
